nixos/tests/systemd-credentials-tpm2.nix

```diff
@@ -0,0 +1,124 @@
+import ./make-test-python.nix ({ lib, pkgs, system, ... }:
+
+let
+  tpmSocketPath = "/tmp/swtpm-sock";
+  tpmDeviceModels = {
```

Comment:
@JJJollyjim would you be interested in backporting the aarch64 fixes into the other tpm-related tests? Maybe we can also move the Tpm class to a more central place, so less needs to be copypasted.